### ISSUE TITLE
Support numbers in references

### ISSUE THREAD

**Issue by danielbotka** [yonaskolb/SwagGen#57]
Hi,

I have a problem with the generator, I get this error message: `Fatal error: Reference #/responses/400 is unresolved: file /SwagGen/Sources/Swagger/Reference.swift, line 10`

The problem is, that I have some references, which are numbers, so imagine the following:
```yaml
'/application/{applicationId}/cob':
    post:
      summary: Lets a user to enter his country of birth
      consumes:
        - application/x-www-form-urlencoded
      parameters:
        - $ref: '#/parameters/applicationId_pathParam'
        - name: cob
          in: formData
          description: Country of birth parameter in ISO 3166-1 alpha-2
          type: string
          required: true
      responses:
        '200':
          description: 'Succesfully added country of birth param to {applicationId}'
        '400':
          $ref: '#/responses/400'
        '500':
          $ref: '#/responses/500'
```
The references are:
```yaml
responses:
  '400':
    description: Validation error
    schema:
      type: array
      items:
        $ref: '#/definitions/BadRequestResponse'
  '404':
    description: Not found
  '500':
    description: Internal error
  '501':
    description: Not implemented yet
  '503':
    description: Service unavailable
```

This is not working. If I modify 400 to Code400 for instance, the generation is working as expected. 

Ps: I also changed the `responses` reference name to `responseCode` reference name as I thought before that the problem is that the `responses` name is special, but the problem persisted, so the problem is the numbers. 

Thanks!!

**Comment by yonaskolb** [comment]
Hi @danielbotka 
Thanks for raising this. At the moment, I don't think response references are actually supported, as per this issue https://github.com/yonaskolb/SwagGen/issues/33.
I'm not sure what you mean when you say `Code400` worked as expected, as that shouldn't either. Can you confirm?

PS You're example swagger specs were very hard to read due to the lack of linkebreaks and whitespace, even though they seem to be in the yaml format. Just something for next time 😉

**Comment by danielbotka** [comment]
Hi @yonaskolb 

Sorry for the format, I've updated it.

So, you're absolutely right, I haven't really checked the generated swift client code itself, I just wanted to be able to generate it for now from the YAML. What I've meant under "work as expected" that if I do something like this:

```yaml
'/application/{applicationId}/cob':
    post:
      summary: Lets a user to enter his country of birth
      consumes:
        - application/x-www-form-urlencoded
      parameters:
        - $ref: '#/parameters/applicationId_pathParam'
        - name: cob
          in: formData
          description: Country of birth parameter in ISO 3166-1 alpha-2
          type: string
          required: true
      responses:
        '200':
          description: 'Succesfully added country of birth param to {applicationId}'
        '400':
          $ref: '#/responses/Code400'
        '500':
          $ref: '#/responses/Code500'
```

And:

```yaml
responses:
  'Code400':
    description: Validation error
    schema:
      type: array
      items:
        $ref: '#/definitions/BadRequestResponse'
  'Code404':
    description: Not found
  'Code500':
    description: Internal error
  'Code501':
    description: Not implemented yet
  'Code503':
    description: Service unavailable
```

Instead of the original I've posted, in this case, I don't get `fatal error` during the generation from the YAML file.

**Comment by yonaskolb** [comment]
It shouldn't be too hard, let me see if I can fix it now

**Comment by yonaskolb** [comment]
Fixed here https://github.com/yonaskolb/SwagGen/pull/58 👍

**Comment by yonaskolb** [comment]
Released in 1.2.0

**Comment by danielbotka** [comment]
Wow, thanks for the prompt fix. It works wonderful!
